py_bind/optv/image_processing.pyx

```diff
@@ -0,0 +1,61 @@
+from optv.parameters cimport ControlParams, control_par
+import numpy as np
+from twisted.cred import error
+cimport numpy as np
+
+def preprocess_image(np.ndarray[ndim=2, dtype=np.uint8_t] input,
+                   int filter_hp,
+                   ControlParams control,
+                   int lowpass_dim=0,
+                   filter_file=None,
+                   np.ndarray[ndim=2, dtype=np.uint8_t] output=None):
+    '''
+    preprocess_image() - perform the steps necessary for preparing an image to 
+    particle detection: an averaging (smoothing) filter on an image, optionally
+    followed by additional user-defined filter.
+    
+    Arguments:
+    numpy.ndarray input - numpy 2d array representing the source image to filter.
+    int filter_hp - flag for additional filtering of _hp. 1 for lowpass, 2 for 
+        general 3x3 filter given in parameter ``filter_file``.
+    ControlParams control - image details such as size and image half for 
+    interlaced cases.
+    int lowpass_dim - dimension of subtracted lowpass image.
```

Comment:
This is the value passed as dim_lp in 
```
int prepare_image(unsigned char * img,
                        unsigned char * img_hp,
                        int dim_lp,
                        int filter_hp,
                        char * filter_file,
                        control_par * cpar)
```
and the C documentation states: "int dim_lp - dimension of subtracted lowpass image.".
So where is the mistake?